Feature/views

Wikitelmex opened the pull request:
# Summary
- modify posts controller, to retrieve correct data
- add user template, to refactor html.erb code
- add posts view index wich fit requirements
- add posts view show wich fit requirements
- modify users show view, wich fit requirements
- fit posts tests
- fit users tests
- improve readme.md

wikitelmex1 committed:
- add users index view
- add user show view
- add validators to users

wikitelmex1 committed:
- modify posts controller, to retrieve correct data
- add user template, to refactor html.erb code
- add posts view index wich fit requirements
- add posts view show wich fit requirements
- modify users show view, wich fit requirements
- fit posts tests
- fit users tests
- improve readme.md

SajjadAhmad14 reviewed (changes requested):
```markdown
Hi Alex, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter checks failing on PR. Kindly make sure all linter checks are passing before asking for a review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

Invalid Code Review Request **does not count** into the code reviews limit._
```

wikitelmex1 committed:
fix stylelint error

wikitelmex1 committed:
linters

wikitelmex1 committed:
removed useless files

rukundoeric commented on app/controllers/posts_controller.rb:
```diff
@@ -1,10 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @user_index = params[:user_id]
+    @user_path = "/users/#{params[:user_id]}"
```
- [x] <b>OPTIONAL:</b> What do you need `@user_path` Every time you create a route, rails generate a method that will return the path of that route. what you have to do is to call `user_path` if it is user, or `post_path` if is post, etc.

Wikitelmex replied:
yeah, I got some issues with that, that I want to prevent just right now

rukundoeric commented on app/controllers/posts_controller.rb:
```diff
@@ -1,10 +1,12 @@
   def index
-    @user_index = params[:user_id]
+    @user_path = "/users/#{params[:user_id]}"
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id])
```
- [x] <b>REQUIRED:</b> There is no need of doing another query here, you can get the post that belongs to that user by simply `@posts = @user.posts`.

rukundoeric commented on app/controllers/posts_controller.rb:
```diff
@@ -1,10 +1,12 @@
   def show
-    @user_index = params[:user_id]
-    @post_index = params[:id]
+    @post = Post.find(params[:id])
+    @comments = Comment.where(post_id: params[:id])
   end
```
- [x] <b>REQUIRED:</b> The same thing goes here, you don't have to make another query. to get the comments that belong to that post, you just do `@comments = @post.comments`.  that's the whole point of the relationships you have created on models of post and comments (`post has_many :comments` and `comment belongs_to :post`).  rails make it so that if there is such a relationship between entities, you can be able to get each other on each other's object. like `@comments = @post.comments` when you need comments that belong to a post or `@post = @comment.post` when you need a post to which that comment belongs.

rukundoeric reviewed (changes requested):
Hi @Wikitelmex,

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [x] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

wikitelmex1 committed:
# Changes required 1
- use implicit @user.posts, instead the query I made before
- use implicit @post.comments, instead the query I made before
- improve routes, using nested routes
- delete @users_path, no longer needed

sinansevgi reviewed (approved):
Hi @Wikitelmex, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Happy Hacking! :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Sinan Sevgi

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
